src/Xdl.Internship.Core/Xdl.Internship.Core.DataAccess/Xdl.Internship.Core.DataAccess.MongoDB/Repositories/MongoRepository.cs

```diff
@@ -0,0 +1,103 @@
+﻿using MongoDB.Bson;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xdl.Internship.Core.DataAccess.MongoDB.Attributes;
+using Xdl.Internship.Core.DataAccess.MongoDB.Entities;
+using Xdl.Internship.Core.DataAccess.MongoDB.Settings;
+
+namespace Xdl.Internship.Core.DataAccess.MongoDB.Repositories
+{
+    public class MongoRepository<TDocument> : IMongoRepository<TDocument> where TDocument : IDocument
+    {
+        private readonly IMongoCollection<TDocument> _collection;
+
+        public MongoRepository(IMongoDbSetting settings)
+        {
+            var database = new MongoClient(settings.ConnectionString).GetDatabase(settings.DatabaseName);
+            _collection = database.GetCollection<TDocument>(GetCollectionName(typeof(TDocument)));
+        }
+
+        private protected string GetCollectionName(Type documentType)
+        {
+            return ((BsonCollectionAttribute)documentType.GetCustomAttributes(
+                    typeof(BsonCollectionAttribute),
+                    true)
+                .FirstOrDefault())?.CollectionName;
+        }
+
+        public virtual IQueryable<TDocument> AsQueryable()
+        {
+            return _collection.AsQueryable();
+        }
+
+        public virtual IEnumerable<TDocument> FilterBy(
+            Expression<Func<TDocument, bool>> filterExpression)
+        {
+            return _collection.Find(filterExpression).ToEnumerable();
+        }
+
+        public virtual IEnumerable<TProjected> FilterBy<TProjected>(
+            Expression<Func<TDocument, bool>> filterExpression,
+            Expression<Func<TDocument, TProjected>> projectionExpression)
+        {
+            return _collection.Find(filterExpression).Project(projectionExpression).ToEnumerable();
+        }
+
+
+        public virtual Task<TDocument> FindOneAsync(Expression<Func<TDocument, bool>> filterExpression)
+        {
+            return Task.Run(() => _collection.Find(filterExpression).FirstOrDefaultAsync());
```

Comment:
why not async pattern? when you expect single result - make logic clear her to check result